src/util/utils.ts

```diff
@@ -205,9 +209,60 @@ export function joinList(strings_: readonly string[], andOrOr: "and" | "or" = "a
     return joined;
 }
 
+/**
+ * Checks if the running Github server is Enterprise server or not
+ * @returns true if not running on GitHub Enterprise Server
+ */
 export function isGhes(): boolean {
     const ghUrl = new URL(
         process.env.GITHUB_SERVER_URL || "https://github.com"
     );
     return ghUrl.hostname.toUpperCase() !== "GITHUB.COM";
 }
+
+/**
+ * The errors messages from octokit HTTP requests can be poor; prepending the status code helps clarify the problem.
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/explicit-module-boundary-types
+export function getBetterHttpError(err: any): Error {
+    const status = err.status;
+    if (status && err.message) {
+        return new Error(`Received status ${status}: ${err.message}`);
+    }
+    return err;
+}
+
+export function assert(value: unknown): asserts value {
+    if (value === undefined) {
+        throw new Error(`${value} must be defined`);
+    }
+}
+
+/**
+ *
+ * @returns Github personal access token provided by the user
+ */
+export function getPat(): string {
```

Comment:
should this return `undefined` if the PAT isn't provided?

Reply:
yeah it should.

Reply:
I know, but looking at this function in isolation I think returning undefined is better semantically than an empty string

Reply:
Okay, will add that